src/drivers/wayland/lv_wayland.c

```diff
@@ -2221,6 +2224,11 @@ static void _lv_wayland_flush(lv_display_t * disp, const lv_area_t * area, unsig
     /* Modify specified area in buffer */
     for(y = area->y1; y <= area->y2; y++) {
         offset = ((area->x1 + (y * hres)) * bpp);
+        if(app->shm_format == WL_SHM_FORMAT_ARGB8888) {
+            for(x = area->x1; x <= area->x2; x++) {
+                lv_color_premultiply((lv_color32_t *)color_p + (x - area->x1));
+            }
```

Comment:
Have less operations in the loop
```suggestion
            int32_t w = lv_area_get_width(area);
            for(x = 0; x < w; x++) {
                lv_color_premultiply((lv_color32_t *)color_p + x);
            }
```

Reply:
Update: I reworked the for loops slightly.